dali/test/python/test_python_function_operator.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import print_function
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import numpy
+import argparse
+import random
+import torchvision.transforms as transforms
+from PIL import Image
+
+images_dirs = [
+    "/data/imagenet/val-jpeg"
+]
+
+
+def resize(image):
+    res = transforms.Resize((300, 300))
+    return numpy.array(res(Image.fromarray(image)))
+
+
+class CommonPipeline(Pipeline):
+    def __init__(self, batch_size, num_threads, device_id, _seed, image_dir):
+        super(CommonPipeline, self).__init__(batch_size, num_threads, device_id, seed=_seed, exec_async=False,
+                                             exec_pipelined=False)
+        self.input = ops.FileReader(file_root=image_dir)
+        self.decode = ops.HostDecoder(output_type=types.RGB)
+        self.resize = ops.PythonFunction(function_id=id(resize))
+
+    def load(self):
+        jpegs, labels = self.input()
+        decoded = self.decode(jpegs)
+        resized = self.resize(decoded)
+        return resized, labels
+
+    def define_graph(self):
+        pass
+
+
+class BasicPipeline(CommonPipeline):
+    def __init__(self, batch_size, num_threads, device_id, seed, image_dir):
+        super(BasicPipeline, self).__init__(batch_size, num_threads, device_id, seed, image_dir)
+
+    def define_graph(self):
+        images, labels = self.load()
+        return images
+
+
+class PythonOperatorPipeline(CommonPipeline):
+    def __init__(self, batch_size, num_threads, device_id, seed, image_dir, function):
+        super(PythonOperatorPipeline, self).__init__(batch_size, num_threads, device_id, seed, image_dir)
+        self.python_function = ops.PythonFunction(function_id=id(function))
```

Comment:
I somehow don't like the idea of forcing our users to write this `function_id=id(...)`. `PythonFunction(function=f)`.
Would it be possible to rename PythonFunction to something else (e.g. `PythonFunctionImpl`) and create a derived class `PythonFunction` by hand (e.g. in pipeline.py), whose constructor would extract the function id?

Reply:
We can do even more - like with TFRecordReader. Provide python wrapper which will call id() internally and it would be transparent for the user.

Reply:
Fixed by the python wrapper